caom2pipe/tests/test_execute_composable.py

```diff
@@ -267,6 +269,52 @@ def test_data_execute(test_config):
             os.rmdir(test_dir)
 
 
+def test_data_execute_v(test_config):
+    test_config.features.supports_latest_client = True
+    test_obs_id = 'test_obs_id'
+    test_dir = os.path.join(tc.THIS_DIR, test_obs_id)
+    test_fits_fqn = os.path.join(test_dir,
+                                 tc.TestStorageName().file_name)
+    try:
+        if not os.path.exists(test_dir):
+            os.mkdir(test_dir, mode=0o755)
+        # precondition = open(test_fits_fqn, 'w')
+        # precondition.close()
+
+        test_data_visitors = [TestVisit]
+        repo_client_mock = Mock(autospec=True)
+        cadc_client_mock = Mock(autospec=True)
+        cadc_client_mock.copy.side_effect = tc.mock_copy_md5
+        test_observer = Mock(autospec=True)
+        test_cred = ''
+        test_transferrer = transfer_composable.VoTransfer()
+        test_transferrer.cadc_client = cadc_client_mock
+
+        ec.CaomExecute._data_cmd_info = Mock(side_effect=_get_fname)
+        repo_client_mock.read.side_effect = tc.mock_read
+
+        # run the test
+        test_executor = ec.DataVisit(
+            test_config, tc.TestStorageName(), test_cred,
+            cadc_client_mock, repo_client_mock, test_data_visitors,
+            mc.TaskType.MODIFY, test_observer, test_transferrer)
+        test_executor.execute(None)
+
+        # check that things worked as expected
+        assert repo_client_mock.read.called, 'read call missed'
+        assert repo_client_mock.update.called, 'update call missed'
+        assert test_observer.metrics.observe.called, 'observe not called'
+        assert cadc_client_mock.copy.called, 'copy not called'
+        cadc_client_mock.copy.assert_called_with(
+            'ad:TEST/test_obs_id.fits.gz', test_fits_fqn, send_md5=True), \
+            'wrong call args'
+    finally:
+        if os.path.exists(test_fits_fqn):
```

Comment:
Alternatively, you could use the `tempfile` module which cleans after itself.

Reply:
All the file naming patterns are encapsulated in the class TestStorageName, which is also a parameter to the class under test (DataVisit), so a temp file name would break the code.

Reply:
`tempfile` allows you to create a directory of temporary files. Any files in there are removed when tests are done.